drivers/scheduler/openshift/openshift.go

```diff
@@ -57,6 +59,70 @@ func (k *openshift) StartSchedOnNode(n node.Node) error {
 	return nil
 }
 
+func (k *openshift) Schedule(instanceID string, options scheduler.ScheduleOptions) ([]*scheduler.Context, error) {
+	var apps []*spec.AppSpec
+	if len(options.AppKeys) > 0 {
+		for _, key := range options.AppKeys {
+			spec, err := k.SpecFactory.Get(key)
+			if err != nil {
+				return nil, err
+			}
+			apps = append(apps, spec)
+		}
+	} else {
+		apps = k.SpecFactory.GetAll()
+	}
+
+	var contexts []*scheduler.Context
+	for _, app := range apps {
+
+		appNamespace := app.GetID(instanceID)
+		k8sOps := k8s_ops.Instance()
+
+		// Update security context for namespace and user
+		if err := k.updateSecurityContextConstraints(k8sOps, appNamespace); err != nil {
+			return nil, err
+		}
+
+		specObjects, err := k.CreateSpecObjects(app, appNamespace, options.StorageProvisioner)
+		if err != nil {
+			return nil, err
+		}
+
+		ctx := &scheduler.Context{
+			UID: instanceID,
+			App: &spec.AppSpec{
+				Key:      app.Key,
+				SpecList: specObjects,
+				Enabled:  app.Enabled,
+			},
+		}
+
+		contexts = append(contexts, ctx)
+	}
+
+	return contexts, nil
+}
+
+func (k *openshift) updateSecurityContextConstraints(k8sOps k8s_ops.Ops, namespace string) error {
```

Comment:
No need to pass `k8sOps`. We could just do `k8s_ops.Instance().GetSCC()` as this is a singleton and we are not creating an object everytime.

Reply:
Done.